File: packages/core-ts/src/routing/flags.ts
/**
 * Feature flags for routing logic.
 *
 * `experimentalRouting` gates the new multi-chain routing model that resolves
 * addresses via `SourceAddress` / `TargetChains`. When `false` (the default)
 * the stable memo-based routing path is used. When `true` the experimental
 * path is invoked instead.
 *
 * The flag can be set per-call through {@link RoutingInput.flags} so that
 * callers can opt in without affecting the rest of the application.
 */
export type RoutingFlags = {
  /** Enable the experimental multi-chain routing path. Defaults to `false`. */
  experimentalRouting?: boolean;
};

/** Default flag values — stable behaviour is always the default. */
export const DEFAULT_ROUTING_FLAGS: Required<RoutingFlags> = {
  experimentalRouting: false,
};

/**
 * Merges caller-supplied flags with the defaults, returning a fully-resolved
 * flag set. Missing flags fall back to their default values.
 */
export function resolveFlags(flags?: RoutingFlags): Required<RoutingFlags> {
  return { ...DEFAULT_ROUTING_FLAGS, ...flags };
}

File: packages/core-go/routing/flags.go
package routing

// RoutingFlags controls optional routing behaviour on a per-call basis.
// All flags default to false (the stable production path).
type RoutingFlags struct {
	// ExperimentalRouting enables the experimental multi-chain routing path.
	// When false (the default) the stable memo-based routing logic is used.
	// When true the experimental path is invoked instead.
	ExperimentalRouting bool `json:"experimentalRouting,omitempty"`
}

// DefaultRoutingFlags returns a RoutingFlags value with all flags set to
// their default (stable) values.
func DefaultRoutingFlags() RoutingFlags {
	return RoutingFlags{
		ExperimentalRouting: false,
	}
}

// ResolveFlags merges the caller-supplied flags with the defaults.
// A nil pointer is treated as "no overrides" — defaults are returned as-is.
func ResolveFlags(f *RoutingFlags) RoutingFlags {
	if f == nil {
		return DefaultRoutingFlags()
	}
	return *f
}
